Failed to analysis

```
java.io.IOException: Bad file descriptor
	at sun.nio.ch.FileDispatcherImpl.size0(Native Method)
	at sun.nio.ch.FileDispatcherImpl.size(FileDispatcherImpl.java:107)
	at sun.nio.ch.FileChannelImpl.size(FileChannelImpl.java:359)
	at sun.nio.ch.FileChannelImpl.transferTo(FileChannelImpl.java:649)
	at shark.FileSourceProvider$openRandomAccessSource$1.read(FileSourceProvider.kt:19)
	at shark.ConstantMemoryMetricsDualSourceProvider$openRandomAccessSource$1.read(ConstantMemoryMetricsDualSourceProvider.kt:58)
	at shark.RandomAccessHprofReader.readRecord(RandomAccessHprofReader.kt:36)
	at shark.HprofHeapGraph.readObjectRecord(HprofHeapGraph.kt:343)
	at shark.HprofHeapGraph.readInstanceDumpRecord$shark_graph(HprofHeapGraph.kt:328)
	at shark.HeapObject$HeapInstance.readRecord(HeapObject.kt:367)
	at shark.internal.PathFinder.readAllNonNullFieldsOfReferenceType(PathFinder.kt:531)
	at shark.internal.PathFinder.visitInstance(PathFinder.kt:477)
	at shark.internal.PathFinder.findPathsFromGcRoots(PathFinder.kt:277)
	at shark.internal.PathFinder.findPathsFromGcRoots(PathFinder.kt:225)
	at shark.HeapAnalyzer.findLeaks(HeapAnalyzer.kt:197)
	at shark.HeapAnalyzer.analyzeGraph(HeapAnalyzer.kt:175)
	at shark.HeapAnalyzer.analyze(HeapAnalyzer.kt:101)
	at leakcanary.internal.HeapAnalyzerService.analyzeHeap(HeapAnalyzerService.kt:87)
	at leakcanary.internal.HeapAnalyzerService.onHandleIntentInForeground(HeapAnalyzerService.kt:61)
	at leakcanary.internal.ForegroundService.onHandleIntent(ForegroundService.kt:55)
	at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:78)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:241)
	at android.os.HandlerThread.run(HandlerThread.java:67)
	Suppressed: java.io.IOException: close failed: EBADF (Bad file descriptor)
		at libcore.io.IoBridge.closeAndSignalBlockedThreads(IoBridge.java:255)
		at java.io.FileInputStream.close(FileInputStream.java:424)
		at sun.nio.ch.FileChannelImpl.implCloseChannel(FileChannelImpl.java:164)
		at java.nio.channels.spi.AbstractInterruptibleChannel.close(AbstractInterruptibleChannel.java:116)
		at shark.FileSourceProvider$openRandomAccessSource$1.close(FileSourceProvider.kt:21)
		at shark.ConstantMemoryMetricsDualSourceProvider$openRandomAccessSource$1.close(ConstantMemoryMetricsDualSourceProvider.kt:63)
		at shark.RandomAccessHprofReader.close(RandomAccessHprofReader.kt:51)
		at shark.HprofHeapGraph.close(HprofHeapGraph.kt:199)
		at kotlin.io.CloseableKt.closeFinally(Closeable.kt:60)
		at shark.HeapAnalyzer.analyze(HeapAnalyzer.kt:98)
		... 7 more
	Caused by: android.system.ErrnoException: close failed: EBADF (Bad file descriptor)
		at libcore.io.Linux.close(Native Method)
		at libcore.io.ForwardingOs.close(ForwardingOs.java:89)
		at libcore.io.BlockGuardOs.close(BlockGuardOs.java:108)
		at libcore.io.ForwardingOs.close(ForwardingOs.java:89)
		at libcore.io.IoBridge.closeAndSignalBlockedThreads(IoBridge.java:253)
		... 16 more

Build.VERSION.SDK_INT: 29
Build.MANUFACTURER: motorola
LeakCanary version: 2.6
Analysis duration: 3098 ms
Heap dump file path: /storage/emulated/0/Download/leakcanary-com.bizzbuzz.bizzbuzz/2021-04-16_16-48-48_700.hprof
Heap dump timestamp: 1618571934968
```

--- a/shark-hprof/src/main/java/shark/FileSourceProvider.kt
+++ b/shark-hprof/src/main/java/shark/FileSourceProvider.kt
@@ -17,7 +17,14 @@ class FileSourceProvider(private val file: File) : DualSourceProvider {
         byteCount: Long
       ) = channel.transferTo(position, byteCount, sink)
 
-      override fun close() = channel.close()
+      override fun close() {
+        try {
+          channel.close()
+        } catch (ignored: Throwable) {
+          // https://github.com/square/leakcanary/issues/2113
+          SharkLog.d(ignored) { "Failed to close channel, ignoring" }
+        }
+      }
     }
   }
 }

